Common/Core/PID/PIDResponse.h

```diff
@@ -42,6 +44,173 @@ void packInTable(const float& separation, tableType& table, const float& lowest,
+using hasTOFAl = decltype(std::declval<T&>().tofNSigmaAl());
+
+template <typename TrackType>
+const auto tofNSigma(const o2::track::PID::ID& index, const TrackType& track)
+{
+  switch (index) {
+    case o2::track::PID::Electron:
+      if constexpr (std::experimental::is_detected<hasTOFEl, TrackType>::value) {
+        return track.tofNSigmaEl();
+      } else {
+        return defaultNSigma;
+      }
+    case o2::track::PID::Muon:
+      if constexpr (std::experimental::is_detected<hasTOFMu, TrackType>::value) {
+        return track.tofNSigmaMu();
+      } else {
+        return defaultNSigma;
+      }
+    case o2::track::PID::Pion:
+      if constexpr (std::experimental::is_detected<hasTOFPi, TrackType>::value) {
+        return track.tofNSigmaPi();
+      } else {
+        return defaultNSigma;
+      }
+    case o2::track::PID::Kaon:
+      if constexpr (std::experimental::is_detected<hasTOFKa, TrackType>::value) {
+        return track.tofNSigmaKa();
+      } else {
+        return defaultNSigma;
+      }
+    case o2::track::PID::Proton:
+      if constexpr (std::experimental::is_detected<hasTOFPr, TrackType>::value) {
+        return track.tofNSigmaPr();
+      } else {
+        return defaultNSigma;
+      }
+    case o2::track::PID::Deuteron:
+      if constexpr (std::experimental::is_detected<hasTOFDe, TrackType>::value) {
+        return track.tofNSigmaDe();
+      } else {
+        return defaultNSigma;
+      }
+    case o2::track::PID::Triton:
+      if constexpr (std::experimental::is_detected<hasTOFTr, TrackType>::value) {
+        return track.tofNSigmaTr();
+      } else {
+        return defaultNSigma;
+      }
+    case o2::track::PID::Helium3:
+      if constexpr (std::experimental::is_detected<hasTOFHe, TrackType>::value) {
+        return track.tofNSigmaHe();
+      } else {
+        return defaultNSigma;
+      }
+    case o2::track::PID::Alpha:
+      if constexpr (std::experimental::is_detected<hasTOFAl, TrackType>::value) {
+        return track.tofNSigmaAl();
+      } else {
+        return defaultNSigma;
+      }
+    default:
+      return defaultNSigma;
+  }
+}
+
+template <class T>
+using hasTPCEl = decltype(std::declval<T&>().tpcNSigmaEl());
+template <class T>
+using hasTPCMu = decltype(std::declval<T&>().tpcNSigmaMu());
+template <class T>
+using hasTPCPi = decltype(std::declval<T&>().tpcNSigmaPi());
+template <class T>
+using hasTPCKa = decltype(std::declval<T&>().tpcNSigmaKa());
+template <class T>
+using hasTPCPr = decltype(std::declval<T&>().tpcNSigmaPr());
+template <class T>
+using hasTPCDe = decltype(std::declval<T&>().tpcNSigmaDe());
+template <class T>
+using hasTPCTr = decltype(std::declval<T&>().tpcNSigmaTr());
+template <class T>
+using hasTPCHe = decltype(std::declval<T&>().tpcNSigmaHe());
+template <class T>
+using hasTPCAl = decltype(std::declval<T&>().tpcNSigmaAl());
+
+template <typename TrackType>
+const auto tpcNSigma(const o2::track::PID::ID& index, const TrackType& track)
```

Comment:
Passing by reference a 1 byte variable is more expensive than passing it by value (it won't be dramatic, but stil...)